### ISSUE TITLE
The logger is sometimes too noisy but not configurable

### ISSUE THREAD

**Issue by Ferdy89** [DataDog/dd-trace-rb#99]
We currently have a big Rails app running on Heroku and using DataDog. There are a few legitimate situations under which we see the following message flooding our consoles and logs:
```
ERROR -- : Connection refused - connect(2) for "127.0.0.1" port 8126
```
Some of these situations are:
* When compiling assets before a deployment. The Datadog buildpack won't start the agent until the application is actually ready to be started, so during asset compilation our console keeps throwing these, one every second.
* Opening up a console. We have configured a "custom" port (the official buildpack still has an agent that listens to port 7777, so we change the port in our Datadog initializer) but that configuration only takes place after initialization. Our application is large enough so that initializing takes a few seconds, during which we see some of those errors popping up in our screen.

Just these two are enough to annoy us very much. We think a simple solution would be to just have the `logger` attribute on `tracer.rb` be an `attr_writer` so that it can be manually configured. We're currently just doing `instance_variable_set('@logger', our_logger)` to pass a `Logger` that outputs to a file log instead of the `STDOUT`.

I would be happy to send a PR if you think this is a reasonable idea. Any thoughts?

**Comment by ufoot** [comment]
Hi, thanks for reporting this. Indeed, being able to override the default `Logger` is something which could be very useful. One thing we also consider doing is prefixing our messages when using the default logger, so that messages coming from `ddtrace` are clearly identified as such (currently, we find identifying them within the flow of a complete app is not convenient).

**Comment by ufoot** [comment]
FYI -> https://github.com/DataDog/dd-trace-rb/pull/102 this should fix your issue I think, introducing a `log=` func to tracer.

**Comment by ufoot** [comment]
#102 merged into master, `0.6.1` about to be released.
